client/my-sites/plans-v2/upsell.tsx

```diff
@@ -1,41 +1,200 @@
+	onPurchaseBothProducts: () => void;
+}
+
+const UpsellComponent = ( {
+	currencyCode,
+	onPurchaseSingleProduct,
+	onPurchaseBothProducts,
+	mainProduct,
+	upsellProduct,
+}: Props ) => {
+	const translate = useTranslate();
+
+	// Upsell prices should come from the API with the upsell products
+	const { originalPrice, discountedPrice } = useItemPrice(
+		upsellProduct,
+		upsellProduct.monthlyProductSlug || ''
+	);
+	const savedAmount = 100;
+
+	const { shortName: mainProductName } = mainProduct;
+	const { shortName: upsellProductName } = upsellProduct;
+
+	const { symbol: currencySymbol } = getCurrencyObject( originalPrice, currencyCode ) || {
+		symbol: '$',
+	};
+
+	return (
+		<Main className="upsell">
+			<div className="upsell__header">
+				<FormattedHeader
+					headerText={ preventWidows(
+						translate( 'Would you like to add %s?', {
+							args: [ upsellProductName ],
+							comment: '%s is the name of a product such as Jetpack Scan or Jetpack Backup',
+						} )
+					) }
+					brandFont
+				/>
+				<div className="upsell__icons">
+					<ProductIcon className="upsell__product-icon" slug={ mainProduct.iconSlug } />
+					<Gridicon className="upsell__plus-icon" icon="plus-small" />
+					<ProductIcon className="upsell__product-icon" slug={ upsellProduct.iconSlug } />
+				</div>
+				<p className="upsell__subheader">
+					{ preventWidows(
+						translate( 'Bundle %(mainProduct)s with %(upsellProduct)s and save!', {
+							args: {
+								mainProduct: mainProductName,
+								upsellProduct: upsellProductName,
+							},
+							comment:
+								'%(mainProduct)s and %(upsellProduct)s are abbreviated name of product such as Scan or Backup',
+						} )
+					) }
+				</p>
+				<p className="upsell__saved-amount">
+					{ preventWidows(
+						translate(
+							'Save %(currencySymbol)s%(savedAmount)d/year on %(upsellProduct)s when paired with %(mainProduct)s',
+							{
+								args: {
+									savedAmount,
+									mainProduct: mainProductName,
+									upsellProduct: upsellProductName,
+									currencySymbol,
+								},
+								comment:
+									'%(savedAmount)s refers to the saved amount by purchasing two products together such as Jetpack Backup and Jetpack Scan',
+							}
+						)
+					) }
+				</p>
+			</div>
+
+			<JetpackProductCard
+				iconSlug={ upsellProduct.iconSlug }
+				productName={ upsellProduct.displayName }
+				subheadline={ upsellProduct.tagline }
+				description={ upsellProduct.description }
+				currencyCode={ currencyCode }
+				billingTimeFrame={ durationToText( upsellProduct.term ) }
+				buttonLabel={ translate( 'Yes, add %s', { args: [ upsellProductName ] } ) }
+				features={ { items: [] } }
+				discountedPrice={ discountedPrice }
+				originalPrice={ originalPrice }
+				onButtonClick={ onPurchaseBothProducts }
+				cancelLabel={ translate( 'No, I do not want %s', {
+					args: [ upsellProductName ],
+				} ) }
+				onCancelClick={ onPurchaseSingleProduct }
+			/>
+		</Main>
+	);
+};
```

Comment:
Good for now, but let's re-use the `ProductCardWrapper` component soon.

Reply:
Will do in follow up PR because I we need to update `ProductCardWrapper` to make it accept a `onCancelClick` and `cancelLabel`.